### PR TITLE
add remove row/column

### PR DESCRIPTION
**Pull request by latot**
Fixes https://github.com/cbm755/octsympy/issues/569

I don't know how we skip this D:

### CODE REVIEW

**Comment by latot** [comment]
Still left 3 cases to this works, let me think how fix it.

**Comment by latot** [comment]
Okidokis, Full Fixed!

File: inst/@sym/subsasgn.m [hunk 1]
```diff
@@ -95,7 +95,42 @@
...
+                  out = subsasgn (out, substruct ('()', {idx.subs{1}}), []);
+                end
+              end
+            case 2
+              if strcmp(idx.subs{1}, ':')
+                out = python_cmd('_ins[0].col_del(_ins[1] - 1); return _ins[0],', val, sym(idx.subs{2}));
+              elseif strcmp(idx.subs{2}, ':')
+                out = python_cmd('_ins[0].row_del(_ins[1] - 1); return _ins[0],', val, sym(idx.subs{1}));
+              else
+                error('A null assignment can only have one non-colon index.'); %% Standard octave error
```
Comment (cbm755) [inline inst/@sym/subsasgn.m]: does this error have BIST?

File: inst/@sym/subsasgn.m [hunk 2]
```diff
@@ -464,4 +499,56 @@
...
+%! assert( isequal( a, sym([])))
+
+%!test
+%! % Dissamble Matrix
+%! a = sym([1, 2; 3, 4; 5 6]);
+%! b = sym([3, 5, 2, 4, 6]);
+%! a(1) = [];
+%! assert( isequal( a, b));
+
+%!error
```
Comment (cbm755) [inline inst/@sym/subsasgn.m]: include some `<keyword>` for the error?

File: inst/@sym/subsasgn.m (hunk 1 above)
Comment (latot) [inline inst/@sym/subsasgn.m]: sorry my ignorance, what is BIST?

**Comment by cbm755** [comment]
Tests look great and we definitely need a fix.  Fix feels long though...

Can it be made inside `mat_replace`?  Or maybe even `mat_rclist_asgn` and `mat_mask_asgn` need to deal with the empty `b` case?  I could be wrong: its so long since I wrote all that spaghetti code!

But I think the intention was the private helpers would do the actual work.

File: inst/@sym/subsasgn.m (hunk 1 above)
Comment (cbm755) [inline inst/@sym/subsasgn.m]: built in self test.  I just mean `%! error <null assignment> ...`

Comment (cbm755) [inline inst/@sym/subsasgn.m]: oops `%!error <null assignment> ...`

File: inst/@sym/subsasgn.m (hunk 2 above)
Comment (latot) [inline inst/@sym/subsasgn.m]: okis, added

**Comment by latot** [comment]
Okis, let me see how this parts works to mix all this things.

**Comment by latot** [comment]
Okidokis, after think this, i move the code to ```mat_replace.m```, the ```mat_mask_asgn``` and ```mat_rclist_asgn```  works with assign of values case by case but ```[]``` its the special case of remove instead of set, so now the code start checking if b is empty, if not continues normally.

Code Updated!

**Comment by latot** [comment]
@cbm755 a little question about Sympy, well you know i'm not an expert of python, but i want request to Sympy the solver for inequalities, actually its very incomplete, can't works with more than one var, it would be useful if i request the feature and in the issue i explain the logic of an inequality solver?

**Comment by cbm755** [comment]
I'm not an expert on Sympy solvers.  There is/was a lot of work going on to improve solvers; I think called something like "solveset"/setsolve.  Search SymPy issues to learn more I guess?

**Comment by latot** [comment]
Oh, sadly there are no issues about inequalities, only about eqs, (i search "ineq" in sympy) and i check the section you recommend D:
i think i'll write it, Thx.

**Pull request merged**